docs.go

```diff
@@ -0,0 +1,289 @@
+	addDetails bool,
+) []string {
+	args := []string{}
+	for _, flag := range flags {
+		modifiedArg := opener
+		for _, s := range strings.Split(flag.GetName(), ",") {
+			trimmed := strings.TrimSpace(s)
+			if len(modifiedArg) > len(opener) {
+				modifiedArg += sep
+			}
+			if len(trimmed) > 1 {
+				modifiedArg += "--" + trimmed
+			} else {
+				modifiedArg += "-" + trimmed
+			}
+		}
+		modifiedArg += closer
+		if flagTakesValue(flag) {
+			modifiedArg += "=" + value
+		}
+
+		if addDetails {
+			modifiedArg += flagDetails(flag)
+		}
+
+		args = append(args, modifiedArg)
+
+	}
+	sort.Strings(args)
+	return args
+}
+
+// flagTakesValue returns true if the flag takes a value, otherwise false
+func flagTakesValue(flag Flag) bool {
+	if _, ok := flag.(BoolFlag); ok {
+		return false
+	}
+	if _, ok := flag.(BoolTFlag); ok {
+		return false
+	}
+	if _, ok := flag.(DurationFlag); ok {
+		return true
+	}
+	if _, ok := flag.(Float64Flag); ok {
+		return true
+	}
+	if _, ok := flag.(GenericFlag); ok {
+		return true
+	}
+	if _, ok := flag.(Int64Flag); ok {
+		return true
+	}
+	if _, ok := flag.(IntFlag); ok {
+		return true
+	}
+	if _, ok := flag.(IntSliceFlag); ok {
+		return true
+	}
+	if _, ok := flag.(Int64SliceFlag); ok {
+		return true
+	}
+	if _, ok := flag.(StringFlag); ok {
+		return true
+	}
+	if _, ok := flag.(StringSliceFlag); ok {
+		return true
+	}
+	if _, ok := flag.(Uint64Flag); ok {
+		return true
+	}
+	if _, ok := flag.(UintFlag); ok {
+		return true
+	}
+	return false
+}
+
+// flagDetails returns a string containing the flags metadata
+func flagDetails(flag Flag) string {
+	description := ""
+	value := ""
+	if f, ok := flag.(BoolFlag); ok {
+		description = f.Usage
+	}
+	if f, ok := flag.(BoolTFlag); ok {
+		description = f.Usage
+	}
+	if f, ok := flag.(DurationFlag); ok {
+		description = f.Usage
+		value = f.Value.String()
+	}
+	if f, ok := flag.(Float64Flag); ok {
+		description = f.Usage
+		value = fmt.Sprintf("%f", f.Value)
+	}
+	if f, ok := flag.(GenericFlag); ok {
+		description = f.Usage
+		if f.Value != nil {
+			value = f.Value.String()
+		}
+	}
+	if f, ok := flag.(Int64Flag); ok {
+		description = f.Usage
+		value = fmt.Sprintf("%d", f.Value)
+	}
+	if f, ok := flag.(IntFlag); ok {
+		description = f.Usage
+		value = fmt.Sprintf("%d", f.Value)
+	}
+	if f, ok := flag.(IntSliceFlag); ok {
+		description = f.Usage
+		if f.Value != nil {
+			value = f.Value.String()
+		}
+	}
+	if f, ok := flag.(Int64SliceFlag); ok {
+		description = f.Usage
+		if f.Value != nil {
+			value = f.Value.String()
+		}
+	}
+	if f, ok := flag.(StringFlag); ok {
+		description = f.Usage
+		value = f.Value
+	}
+	if f, ok := flag.(StringSliceFlag); ok {
+		description = f.Usage
+		if f.Value != nil {
+			value = f.Value.String()
+		}
+	}
+	if f, ok := flag.(Uint64Flag); ok {
+		description = f.Usage
+		value = fmt.Sprintf("%d", f.Value)
+	}
+	if f, ok := flag.(UintFlag); ok {
+		description = f.Usage
+		value = fmt.Sprintf("%d", f.Value)
+	}
```

Comment:
⭐⭐⭐ _(must change)_

I'm _pretty sure_ these various conditions for flags can be represented as a single if statement. So like

```golang
		description = f.Usage
		if f.Value != nil {
		    value = fmt.Sprintf("%f", f.Value)
		}
```

So that we can avoid the cognitive overhead of the type checks.

Reply:
As above , I added a `GetUsage() string` and `GetValue() string` method to the `Flag` interface and removed the type assertions completely.